src/server/sams/api/produce/assets.py

```diff
@@ -68,6 +82,20 @@ def update(self, id, updates, original):
         """
         Uses 'id' and updates the corresponding asset
         """
+        # Get set_id for asset using internal assets service
+        assets_service = get_assets_service()
+        set_id = assets_service.get_by_id(id).get('set_id')
```

Comment:
This should not be required, as this function receives `id`, `updates` and `original`. You could use `original.get('set_id')` instead